src/OnlineRetailPortal.Core/Models/Product.cs

```diff
@@ -1,25 +1,70 @@
-﻿using System;
+﻿using OnlineRetailPortal.Contracts;
+using System;
 using System.Collections.Generic;
 using System.Linq;
 using System.Text;
-namespace OnlineRetailPortal.Core.Models
+using System.Threading.Tasks;
+
+namespace OnlineRetailPortal.Core
 {
     public class Product
     {
         public string Id { get; set; }
         public string Name { get; set; }
-        public string Description { get; set; }
-        public Image HeroImage { get; set; }
         public Price Price { get; set; }
+        public Price IsNegotiable { get; set; }
+        public Price Currency { get; set; }
         public Category Category { get; set; }
-        public Status Status { get; set; }
+        public Image HeroImage { get; set; }
+        public Address PickupAddress { get; set; }
+        public List<Image> Images { get; set; }
+        public string UserId { get; set; }
         public DateTime PostDateTime { get; set; }
+        public DateTime PurchasedDate { get; set; }
+        public string Description { get; set; }
         public DateTime ExpirationDate { get; set; }
 
-        public List<Image> Images { get; set; }
-        public DateTime PurchasedDate { get; set; }
-        public Address PickupAddress { get; set; }
+        public Status Status { get; set; }
+
+        IProductStore productStore;
+
+        public Product()
```

Comment:
remove this if its not having any  references

Reply:
it is for mock db